convert2rhel/unit_tests/actions/system_checks/convert2rhel_latest_test.py

```diff
@@ -56,9 +56,9 @@ class TestCheckConvert2rhelLatest:
     @pytest.mark.parametrize(
         ("convert2rhel_latest_version_test",),
         (
-            [{"local_version": "0.20", "package_version": "C2R convert2rhel-0:0.22-1.el7.noarch", "pmajor": "7"}],
-            [{"local_version": "0.20", "package_version": "C2R convert2rhel-0:0.22.0-1.el7.noarch", "pmajor": "7"}],
-            [{"local_version": "0.20.0", "package_version": "C2R convert2rhel-0:0.22-1.el7.noarch", "pmajor": "7"}],
+            [{"local_version": "0.20", "package_version": "conver2rhel-0:0.22-1.el7.noarch", "pmajor": "7"}],
+            [{"local_version": "0.20", "package_version": "conver2rhel-0:0.22.0-1.el7.noarch", "pmajor": "7"}],
+            [{"local_version": "0.20.0", "package_version": "conver2rhel-0:0.22-1.el7.noarch", "pmajor": "7"}],
```

Comment:
@r0x0d Do we still need C2R here?

Reply:
Short answer is yes, we still need that.

Longer answer is that package_version is being set as the output of utils.run_subprocess.  But we now call run_subsprocess multiple times.  So we need to have multiple outputs and we need to mock it so that it gives different output each time.

We can call it up to three times:
* repoquery
* rpm -qf
* rpm -V

Reply:
Sorry for the late reply, @SpyTec, but yes. 

We are using the C2R identifier to filter out the output and make sure we are only parsing the results that contain the C2R identifier and not breaking out the parser with any other message that may come from repoquery.

```python
        # We are expecting an repoquery output to be similar to this:
        # C2R convert2rhel-0:0.17-1.el7.noarch
        # We need the `C2R` identifier to be present on the line so we can know for
        # sure that the line we are working with is the a line that contains
        # relevant repoquery information to our check, otherwise, we just log the
        # information as debug and do nothing with it.
        for raw_version in raw_output_convert2rhel_versions:
            if "C2R" in raw_version:
                temp_raw_output.append(raw_version.lstrip("C2R "))
            else:
                # Mainly for debugging purposes to see what is happening if we got
                # anything else that does not have the C2R identifier at the start
                # of the line.
                logger.debug("Got a line without the C2R identifier: %s" % raw_version)
```

You can see that we have a minimal parser for this line, where if we identify the C2R id, we know its a line we can parse, otherwise, it is a message that repoquery outputs, and we want to log it as debug.